zerver/views/previewable.py

```diff
@@ -0,0 +1,29 @@
+import re
+
+from django.http import HttpRequest, HttpResponse
+from django.utils.translation import gettext as _
+
+from zerver.lib.exceptions import JsonableError
+from zerver.lib.github import get_issue_or_pr_data
+from zerver.lib.request import REQ, has_request_variables
+from zerver.lib.response import json_success
+from zerver.models import UserProfile
+
+
+@has_request_variables
+def get_previewable_data(
+    request: HttpRequest, user_profile: UserProfile, url: str = REQ()
+) -> HttpResponse:
+    github_re = r"(?:(?:https?:\/\/)?(?:www.)?)?github.com\/([^\/]*)\/([^\/]*)\/(issues|pull)\/([^\/|^?|#]*)"
+    github_match = re.match(github_re, url)
```

Comment:
Same comment on parsing here; we should let `urrlib` parse the library and then just compare the hostname to github.com, for example. This is much safer/more secure and easier to read too :)

Reply:
@timabbott, but how we will be knowing if it is a user/org URL or an Issue/PR URL?
Same doubt in the above comment for `util.js`.

Reply:
We'll still need to look at the URL's `path` component, probably with a regex -- i.e. the part that isn't the domain. What I'm trying to do is have the `https` and `github.com` parts use a standard URL library and then just check "is it an https URL {"github.com", "www.github.com"} with clean JS given that's what we mean. The last bit of just parsing the path will still need to involve a regex, of course, but it'll just be on the URL's path -- the `?` parameters, `#` fragments, and other complexity will be handled for us.